lib/shared.bash

```diff
@@ -5,16 +5,33 @@ function plugin_read_list() {
   local prefix="BUILDKITE_PLUGIN_MONOREPO_DIFF_$1"
   local suffix="$2"
   local parameter="${prefix}_0_${suffix}"
+  local chained_parameter="${prefix}_0_${suffix}_0"
 
-  if [[ -n "${!parameter:-}" ]]; then
+  # Reads nested list of multiple watch paths
+  if [[ -n "${!parameter:-}" || -n "${!chained_parameter:-}" ]]; then
     local i=0
-    local parameter="${prefix}_${i}_${suffix}"
-    while [[ -n "${!parameter:-}" ]]; do
-      echo "${!parameter}"
+    while [[ -n "${!parameter:-}" || -n "${!chained_parameter:-}" ]]; do
+      local parameter="${prefix}_${i}_${suffix}"
+      # check for single path oldschool
+      if [[ -n "${!parameter:-}" ]]; then
+      	 echo "${!parameter}"
+      else
+        # check for new style
+        local params=()
+        local j=0
+	      while [[ -n "${!chained_parameter:-}" ]]; do
+		      params+=("${!chained_parameter}")
+		      j=$((j+1))
+		      chained_parameter="${prefix}_${i}_${suffix}_${j}"
+	      done
+        echo "${params[@]}"
+      fi
       i=$((i+1))
-      parameter="${prefix}_${i}_${suffix}"
```

Comment:
I think we actually still need this incase the next in the index list is an old style. You can ditch line 14 once this is back in.